File: BotsSDK/korebot/src/main/java/kore/botssdk/audiocodes/webrtcclient/Structure/ImageViewWithText.java
package kore.botssdk.audiocodes.webrtcclient.Structure;

import android.content.Context;
import android.content.res.TypedArray;
import android.graphics.Typeface;
import android.graphics.drawable.Drawable;
import android.util.AttributeSet;
import android.view.View;
import android.widget.ImageView;
import android.widget.RelativeLayout;
import android.widget.TextView;

import kore.botssdk.R;

public class ImageViewWithText  extends RelativeLayout {
    private static final String TAG = "ImageViewWithText";

    View rootView;
    TextView valueTextView;
    ImageView valueImageView;

    public ImageViewWithText(Context context) {
        super(context);
        init(context, null);
    }

    public ImageViewWithText(Context context, AttributeSet attrs) {
        super(context, attrs);
        init(context, attrs);
    }

    private void init(Context context, AttributeSet attrs) {
        rootView = inflate(context, R.layout.image_view_with_text, this);
        valueTextView = rootView.findViewById(R.id.imageviewwithtext_text);
        valueImageView = rootView.findViewById(R.id.imageviewwithtext_imageview);
        if(attrs!=null)
        {
//            int[] set = {
//                    android.R.attr.background, // idx 0
//                    android.R.attr.src, // idx 1
//                    android.R.attr.text,        // idx 2
//                    //android.R.attr.textColor,       // idx 3
//                    //android.R.attr.text,
//            };
//            TypedArray typedArray = context.obtainStyledAttributes(attrs, set);

            TypedArray typedArray = context.getTheme().obtainStyledAttributes(
                    attrs,
                    R.styleable.ImageViewWithTextStyle,
                    0, 0
            );

            Drawable attBackDrawable = typedArray.getDrawable(R.styleable.ImageViewWithTextStyle_BackgroundImage);
            Drawable attSrcDrawable = typedArray.getDrawable(R.styleable.ImageViewWithTextStyle_srcImage);
            CharSequence attText = typedArray.getText(R.styleable.ImageViewWithTextStyle_Text);
            int textColor = typedArray.getColor(R.styleable.ImageViewWithTextStyle_TextColor, -2);
//            float attTextSize = typedArray.getDimensionPixelSize(R.styleable.ImageViewWithTextStyle_TextSize,-1);


            if(attBackDrawable!=null) {
                valueImageView.setBackground(attBackDrawable);
            }
            if(attSrcDrawable!=null) {
                valueImageView.setImageDrawable(attSrcDrawable);
            }
            if(attText!=null) {
                valueTextView.setText(attText);
            }
            if(textColor!=-2) {
                valueTextView.setTextColor(textColor);
            }
                valueTextView.setTypeface(Typeface.defaultFromStyle(Typeface.BOLD));
//            if(attTextSize>0) {
//                valueTextView.setTextSize(attTextSize);
//            }
            typedArray.recycle();

            //valueTextView.setText(attrs.getAttributeValue(android.R.attr.text));
        }
    }

    public TextView getValueTextView() {
        return valueTextView;
    }

    public ImageView getValueImageView() {
        return valueImageView;
    }
//        minusButton = rootView.findViewById(R.id.minusButton);
//        plusButton = rootView.findViewById(R.id.plusButton);
//
//        minusButton.setOnClickListener(new View.OnClickListener() {
//            @Override
//            public void onClick(View v) {
//                decrementValue(); //we'll define this method later
//            }
//        });
//
//        plusButton.setOnClickListener(new View.OnClickListener() {
//            @Override
//            public void onClick(View v) {
//                incrementValue(); //we'll define this method later        }
//            });
//        }

}

File: BotSdk/ui/src/main/java/com/kore/ui/audiocodes/webrtcclient/structure/ImageViewWithText.kt
package com.kore.ui.audiocodes.webrtcclient.structure

import android.content.Context
import android.graphics.Typeface
import android.util.AttributeSet
import android.view.View
import android.widget.ImageView
import android.widget.RelativeLayout
import android.widget.TextView
import com.kore.ui.R

class ImageViewWithText : RelativeLayout {
    private var rootView: View? = null
    var valueTextView: TextView? = null
    private var valueImageView: ImageView? = null

    constructor(context: Context) : super(context) {
        init(context, null)
    }

    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs) {
        init(context, attrs)
    }

    private fun init(context: Context, attrs: AttributeSet?) {
        rootView = inflate(context, R.layout.image_view_with_text, this)
        valueTextView = rootView?.findViewById(R.id.imageviewwithtext_text)
        valueImageView = rootView?.findViewById(R.id.imageviewwithtext_imageview)
        if (attrs != null) {
            val typedArray = context.theme.obtainStyledAttributes(
                attrs, R.styleable.ImageViewWithTextStyle, 0, 0
            )
            val attBackDrawable = typedArray.getDrawable(R.styleable.ImageViewWithTextStyle_BackgroundImage)
            val attSrcDrawable = typedArray.getDrawable(R.styleable.ImageViewWithTextStyle_srcImage)
            val attText = typedArray.getText(R.styleable.ImageViewWithTextStyle_Text)
            val textColor = typedArray.getColor(R.styleable.ImageViewWithTextStyle_TextColor, -2)
            //            float attTextSize = typedArray.getDimensionPixelSize(R.styleable.ImageViewWithTextStyle_TextSize,-1);
            if (attBackDrawable != null) {
                valueImageView?.background = attBackDrawable
            }
            if (attSrcDrawable != null) {
                valueImageView?.setImageDrawable(attSrcDrawable)
            }
            if (attText != null) {
                valueTextView?.text = attText
            }
            if (textColor != -2) {
                valueTextView?.setTextColor(textColor)
            }
            valueTextView?.typeface = Typeface.defaultFromStyle(Typeface.BOLD)
            typedArray.recycle()
        }
    }

    companion object {
        private const val TAG = "ImageViewWithText"
    }
}